cmd/oras/root/pull_test.go

```diff
@@ -41,3 +44,86 @@ func Test_runPull_errType(t *testing.T) {
 		t.Fatalf("got %v, want %v", got, want)
 	}
 }
+
+func Test_doPull(t *testing.T) {
+	t.Run("PreservePermissions", func(t *testing.T) {
+		repo := "pp"
+		tagReference := "dppp"
+		// Create a temporary directory, put a file in there with mode 0777
+		srcRoot, err := os.MkdirTemp("", "doPull_preservePermissions_srcDir-*")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() { _ = os.RemoveAll(srcRoot) }()
+
+		srcDir, err := os.MkdirTemp(srcRoot, "*")
+		if err != nil {
+			t.Fatal(err)
+		}
+		tf, err := os.OpenFile(fmt.Sprintf("%s/foo.sh", srcDir), os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0777)
+		if err != nil {
+			t.Fatal(err)
+		}
+		oldFs, err := tf.Stat()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		oldWd, err := os.Getwd()
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() { _ = os.Chdir(oldWd) }()
+
+		//fmt.Println(filepath.Base(srcDir))
+		subDir := filepath.Base(srcDir)
+
+		pushArgs := []string{
+			"push",
+			"--insecure",
+			"--plain-http",
+			"--no-tty",
+			"--format",
+			"go-template=\n",
```

Comment:
Do we need the formatting?

Reply:
I couldn't find a way to pass an empty value here for the template, this felt like the easiest way to squash the output when running as a test.

If you dont pass the formatting you get the full output, but maybe that's ok? Ill update this to remove it